syz-manager/hub.go

```diff
@@ -73,17 +73,24 @@ type HubConnector struct {
 type HubManagerView interface {
 	getMinimizedCorpus() (corpus, repros [][]byte)
 	addNewCandidates(candidates []rpctype.Candidate)
+	hubIsUnreachable()
 }
 
 func (hc *HubConnector) loop() {
 	var hub *rpctype.RPCClient
-	for ; ; time.Sleep(10 * time.Minute) {
+	var doneOnce bool
+	for attempt := 0; ; time.Sleep(10 * time.Minute) {
 		corpus, repros := hc.mgr.getMinimizedCorpus()
 		hc.newRepros = append(hc.newRepros, repros...)
 		if hub == nil {
+			attempt++
 			var err error
 			if hub, err = hc.connect(corpus); err != nil {
 				log.Logf(0, "failed to connect to hub at %v: %v", hc.cfg.HubAddr, err)
+				const maxAttempts = 3
+				if attempt >= maxAttempts && !doneOnce {
+					hc.mgr.hubIsUnreachable()
```

Comment:
Will this trigger if connects always succeed, but syncs always fail?

Reply:
Indeed, it won't. I've restructured the code once more.